### PR TITLE
Feat "Added Toast Notifications for Form Submissions in the both The Contact Section and Newsletter"

### PR DESCRIPTION
**Pull request by mosespace**
**Description:**
This pull request enhances both the contact section and the newsletter section which are seen on the home page by implementing toast notifications using the `react-toastify` library. Toast notifications are displayed to provide real-time feedback to users when they submit the contact form. 

**Changes Made:**
- Integrated the `react-toastify` library into the project.
- Created a function to display success and error toast messages.
- Modified the form submission handling logic to trigger toast messages when the forms are submitted successfully or encounters an error.
- Improved the user experience by providing instant feedback on the outcome of the form submission.

**Testing:**
- Tested the form submission functionality and toast notifications under various scenarios, including successful submissions and error cases.
- Ensured that the toast notifications are displayed as expected.

**Screenshots:**
> The Contact Section
image

> The Newsletter
image

### CODE REVIEW

**Comment by mosespace** [comment]
@HassanBahati Kindly Review This

**Comment by Amanya-Brian** [comment]
This is good! Thank you @mosespace

**Comment by mosespace** [comment]
> This is good! Thank you @mosespace

Thank you!

**Review by HassanBahati — COMMENTED** [review]
Looks great @mosespace

can we confirm the information is submitted to the email?

ps: you could try to submit , then check your email to affirm if you authored an email to the recipient ....

**Review by HassanBahati — COMMENTED** [review]
On the other hand...this pr doesnt seem to deploy....

**Comment by HassanBahati** [comment]
> On the other hand...this pr doesnt seem to deploy....

@Amanya-Brian could you please manually trigger a re-deploy of this pr...the build seems to succeed...
image

File: components/home/sections/newsletter.js [hunk 1]
```diff
@@ -1,39 +1,58 @@
...
+import "react-toastify/dist/ReactToastify.css";
 
 export default function Newsletter() {
   const [email, setEmail] = useState("");
 
-  function HandleSubmit(e) {
-    window.location.href = `mailto:pycon-uganda+subscribe@googlegroups.com?subject=Subscribe%20to%20PyCon%20Uganda%20Newsletter`;
-  }
-
+  // function HandleSubmit(e) {
```
Comment (HassanBahati) [inline components/home/sections/newsletter.js:9]:
```suggestion
  // function HandleSubmit(e) {
```

The dead code could be removed...

File: components/home/sections/contact.js [hunk 2]
```diff
@@ -1,98 +1,118 @@
+"use client";
 import { PYCONUG_EMAIL } from "@/utils/constants";
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
```
Comment (HassanBahati) [inline components/home/sections/contact.js:4]: not really necessary to keep calling `import "react-toastify/dist/ReactToastify.css";` on every usecase....can be configured to apply styles globally...

```suggestion
 ```

File: components/home/sections/newsletter.js [hunk 3]
```diff
@@ -1,39 +1,58 @@
 "use client";
-
 import { useState } from "react";
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
```
Comment (HassanBahati) [inline components/home/sections/newsletter.js:4]: not really necessary to keep calling import "react-toastify/dist/ReactToastify.css"; on every usecase....can be configured to apply styles globally...

```suggestion
 
```

File: components/home/sections/contact.js [hunk 4]
```diff
@@ -1,98 +1,118 @@
...
-              className="hover:bg-pyconug-lightBlue border-2 hover:text-white border-pyconug-lightBlue py-2 px-10 m-2  rounded-3xl"
+              type='submit'
+              className='hover:bg-pyconug-lightBlue border-2 hover:text-white border-pyconug-lightBlue py-2 px-10 m-2  rounded-3xl'
             >
               Send message
             </button>
           </form>
         </div>
       </div>
+      <ToastContainer />
```
Comment (HassanBahati) [inline components/home/sections/contact.js:115]: we could refactor to have the provider instantiated in the `root layout` than calling it on every use case

File: components/home/sections/contact.js [hunk 5]
```diff
@@ -1,98 +1,118 @@
...
+    e.preventDefault();
+    try {
+      toast.success("Your email has been sent successfully 🤗", {
+        position: "bottom-center",
+        theme: "dark",
+      });
+    } catch (error) {
+      console.log(error);
+      toast.error("There was a problem sending the email", {
+        position: "bottom-center",
```
Comment (HassanBahati) [inline components/home/sections/contact.js:17]: for consistency...we could have the configurations made globally while setting  up the provider...than setting properties eg. `position` on every use case....i think we expect same position on every use case

Comment (mosespace) [reply components/home/sections/contact.js:17]: Ive made alll the configurations to be on the ToastContainer which i think resolves this conversation

File: components/home/sections/newsletter.js (hunk 1 above)
Comment (mosespace) [reply components/home/sections/newsletter.js:9]: I've removed the dead code

File: components/home/sections/newsletter.js (hunk 3 above)
Comment (mosespace) [reply components/home/sections/newsletter.js:4]: Yah, i think it's not really necessary. Thank you @HassanBahati

File: components/home/sections/contact.js (hunk 2 above)
Comment (mosespace) [reply components/home/sections/contact.js:4]: Okay, so let me wrapp it in the layout of the app which i think resolves this message

**Comment by mosespace** [comment]
> On the other hand...this pr doesnt seem to deploy....

I think someone has to approval this pull request! the main branch just seems to be protected!.

File: components/home/sections/contact.js (hunk 4 above)
Comment (HassanBahati) [reply components/home/sections/contact.js]: hi @mosespace sorry i think you got me wrong...by `we could refactor to have the provider instantiated in the root layout than calling it on every use case` i didnt mean making the `root layout` a `client component`.....
i meant we could have a provider.... that is called in the root layout, the provider can be a client component..... server components can render client components...
so all notifications configs can be applied through a provider globally....

**Comment by HassanBahati** [comment]
> > On the other hand...this pr doesnt seem to deploy....
> 
> I think someone has to approval this pull request! the main branch just seems to be protected!. Because if i also run i get a successful response
> 
> ```js
> npm run build
> ```

true...it builds fine....though prs should also deploy independent of an approval....

**Comment by HassanBahati** [comment]
> hi @mosespace sorry i think you got me wrong...by `we could refactor to have the provider instantiated in the root layout than calling it on every use case` i didnt mean making the `root layout` a `client component`..... i meant we could have a provider.... that is called in the root layout, the provider can be a client component..... server components can render client components... so all notifications configs can be applied through a provider globally....

I think once we have this...all should be fine

**Comment by Amanya-Brian** [comment]
@HassanBahati @mosespace  I have manually triggered the build

**Comment by HassanBahati** [comment]
> @HassanBahati @mosespace I have manually triggered the build

thank you @Amanya-Brian .... now deploys successfully

**Comment by HassanBahati** [comment]
other thoughts on this...clearing the current form fields once the success message comes 
image

**Comment by HassanBahati** [comment]
any progress @mosespace

**Comment by Amanya-Brian** [comment]
@mosespace did you try to clear the fields after submission?

**Comment by mosespace** [comment]
Hello @HassanBahati and @Amanya-Brian, am sorry for the delay but I've managed to work upon the assigned task of **Clearing The Form After Submission**
Hope this solves the issue.

**Comment by Amanya-Brian** [comment]
@mosespace this is good. Kindly update the newsletter section as well and we close this off

**Comment by mosespace** [comment]
> @mosespace this is good. Kindly update the newsletter section as well and we close this off

Hi, I've also updated the newsletter. Kindly review. Thanks

**Review by Amanya-Brian — APPROVED** [review]
Thanks

**Pull request merged**